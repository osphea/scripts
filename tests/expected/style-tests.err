
Usage: style::stylize <command> [style options] [command parameters]

<command> is any command with output to stylize, followed by style options,
and then the command's normal parameters.

style options include:
  --bold, --faint, --underline, etc.
  --color <color_name>
  --background <color_name>
  --indent <spaces_count>
  --stderr (output to standard error instead of standard out)

  echo "This is $(style::echo -f --bold LOUD) and soft."

command parameters are those supported by the stylized command.

Use style::stylize --help colors for a list of colors or backgrounds
Use style::stylize --help attributes for a list of style attribute flags

Usage: style::printf [style options] [command parameters]

style options include:
  --bold, --faint, --underline, etc.
  --color <color_name>
  --background <color_name>
  --indent <spaces_count>
  --stderr (output to standard error instead of standard out)

  echo "This is $(style::echo -f --bold LOUD) and soft."

command parameters are those supported by the 'printf' command.

Use style::printf --help colors for a list of colors or backgrounds
Use style::printf --help attributes for a list of style attribute flags

Usage: style::error [style options] [command parameters]

Default style options for style::error:
  [1;91m"--stderr --bold --color red"[0m

style options include:
  --bold, --faint, --underline, etc.
  --color <color_name>
  --background <color_name>
  --indent <spaces_count>
  --stderr (output to standard error instead of standard out)

  echo "This is $(style::echo -f --bold LOUD) and soft."

command parameters are those supported by the echo command.

Use style::error --help colors for a list of colors or backgrounds
Use style::error --help attributes for a list of style attribute flags

Usage: style::link [style options] [command parameters]

Default style options for style::link:
  [4;34m"--underline --color dark_blue"[0m

style options include:
  --bold, --faint, --underline, etc.
  --color <color_name>
  --background <color_name>
  --indent <spaces_count>
  --stderr (output to standard error instead of standard out)

  echo "This is $(style::echo -f --bold LOUD) and soft."

command parameters are those supported by the echo command.

Use style::link --help colors for a list of colors or backgrounds
Use style::link --help attributes for a list of style attribute flags
black
blue
cyan
dark_blue
dark_cyan
dark_green
dark_magenta
dark_red
dark_yellow
default
gray
green
light_gray
magenta
pink
purple
red
white
yellow
--blink
--bold
--faint
--italic
--reset
--underline
[1;32mINFO: Info here[0m
[1;33mWARNING: Watch out![0m
[1;91mERROR: What went wrong now?[0m
[5;33mWARNING: Customized warning style, still to stderr! :-)[0m

This should still display in bold red, but on stderr
