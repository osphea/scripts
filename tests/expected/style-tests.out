[94m------------------[0m
[94m------------------[0m
[94m------------------[0m
[94m----- colors -----[0m
[94m--- attributes ---[0m
[94m------------------[0m
[1mstyle::echo --bold[0m
[1;96mstyle::echo --bold --color cyan[0m
[2;92mstyle::echo --faint --color green[0m
[3;95mstyle::echo --italic --color magenta[0m
[4;34mstyle::echo --underline --color dark_blue[0m
[5;37mstyle::echo --blink --color light_gray[0m
[95;46mstyle::echo --pink --background dark_cyan[0m
[3;35mitalic this style may not work in some terminals: style::echo --italic --dark_magenta italic this style may not work in some terminals:[0m
[1m      Item   Cost[0m
[m      ----   ----[0m
[35;107m     beans $  2.90[0m
[35;107m    franks $  9.35[0m
[35;107m      cola $  7.99[0m
[35;107m  tiramasu $ 24.50[0m
[43;30mNow is the time for all good[0m
[43;30mpeople to come to the[0m
[43;30maid of their country and world.[0m
[106;30m    Now is the time for all good[0m
[106;30m    people to come to the[0m
[106;30m    aid of their country and world.[0m
[4;34mhttp://wikipedia.com[0m
STYLE_TO_TTY_ONLY=true

This will not be styled. It doesn't print directly to the tty

This will not be styled. stderr doesn't print directly to the tty
[1;91mSTYLE_TO_TTY_ONLY=false[0m
[96;2m[0m
[96;2mThis will be styled even though it doesn't print directly to the tty.[0m
[96;2m[0m
[96;2mThis will be styled even though stderr doesn't print directly to the tty[0m
This is [m-f --bold --yellow LOUD[0m and soft.
This is [m--force --bold --yellow LOUD[0m and soft.
This is [m--tty --bold --yellow LOUD[0m and soft.
Bad style, Error status: 2
No orange! Error status: 2
